### DISCUSSION TITLE
Show security key on autoregistration page

### DISCUSSION THREAD

**Discussion by ctmifune** [neos-sdi/adfsmfa#324] [Q&A] [answered]
Hi!
Have 2 questions:
1. Can i(how) change autoregistration page to show security key, not only QR-code without do my build? 
    Now i can see QR-code and part of sucurity key, hide other part " (truncated for security reasons)". 
   <img width="283" alt="Screenshot 2024-02-02 141043" src="https://github.com/neos-sdi/adfsmfa/assets/74378776/7c4f7c7a-5624-4904-b267-b4631e5ea2e1">
2. Can system add user email from AD automaticly, after user autoregistration only with TOTP? 
    If i import users from add, system copy AD-attribute "mail" to "msDS-cloudExtensionAttribute11", but user cant do autoregistration.
    And if user do autoregistration only with TOTP(mail not required), field email is empty.

**Comment by redhook62** [comment, marked as answer]
Hi @ctmifune 

1)Yes, you can send the details of the key and QRCode by email.
Your user must have a valid email.

2) Yes, when importing from ADDS

You have 2 solutions:
In the MMC, access your user, view their properties and choose to send the key by email.
When importing from ADDS please specify the -SendMail and possibly -NewKey parameters on the PS command
In any case, your users must have a valid email, not subject to MFA

regards
